habitat/tasks/vln/vln.py

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+from typing import Any, Dict, List, Optional
+
+import attr
+from gym import spaces
+
+from habitat.core.registry import registry
+from habitat.core.simulator import Observations, Sensor
+from habitat.core.utils import not_none_validator
+from habitat.tasks.nav.nav import (
+    NavigationEpisode,
+    NavigationGoal,
+    NavigationTask,
+)
+
+
+@attr.s(auto_attribs=True)
+class InstructionData:
+    instruction_text: str
+    instruction_tokens: Optional[List[str]] = None
+
+
+@attr.s(auto_attribs=True, kw_only=True)
+class VLNEpisode(NavigationEpisode):
+    r"""Specification of episode that includes initial position and rotation of
+    agent, goal specifications, instruction specifications, and optional shortest paths.
+
+    Args:
+        scene_id: id of scene inside the simulator.
+        start_position: numpy ndarray containing 3 entries for (x, y, z).
+        start_rotation: numpy ndarray with 4 entries for (x, y, z, w)
+            elements of unit quaternion (versor) representing agent 3D
+            orientation.
+        instruction: single instruction guide to goal.
+        trajectory_id: id of ground truth trajectory path.
+        goals: relevant goal object/room.
+    """
+    path: List[List[float]] = attr.ib(
```

Comment:
Can we save `path` as `shortest_path` in NavigationEpisode? Also doc string is missing for `path`.

Reply:
The path defined in the R2R dataset is a sequence of [x, y, z] positions. The `shortest_paths` in NavigationEpisode uses `ShortestPathPoint` which requires a rotation to be specified as well. To reconcile, we could make the rotation in `ShortestPathPoint` optional or keep a separate `path` as we have now. Any suggestions on handling that?

Reply:
Make the rotation in `ShortestPathPoint` optional sounds good. You can re-use visualization we have for shortest path already.

Reply:
Probably makes more sense to name this reference path (I believe that is what it is called in the VLN dataset).  That'll keep the shortest-path field open for an oracle trajectory for imitation learning.

Reply:
Agreed, `reference_path` then.